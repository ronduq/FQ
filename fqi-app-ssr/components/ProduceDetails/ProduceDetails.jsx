import React from 'react';
import PropTypes from 'prop-types';
import get from 'lodash.get';
import classNames from 'classnames';
import YouTube from 'react-youtube';
import Card from '../Card/CardContainer';
import s from './ProduceDetails.scss';


const ProduceDetails = ({
  selectedRetailerId,
  retailersData,
  producesData,
  produceInfo,
  viewport,
  content,
}) => {
  const produceCode = produceInfo.info.produceCode;
  const youtubeId = producesData.itemsVideos[produceCode];
  const opts = {
    width: '100%',
    height: 'auto',
    playerVars: { // https://developers.google.com/youtube/player_parameters
      autoplay: 0,
      rel: 0,
      showInfo: 0,
      modestbranding: 1,
      controls: 1,
      frameBorder: 0,
    }
  };
  return (
    <div className={s.container}>
      <h3 className={s.title}>{content.produce_metrics}</h3>
      { Object.entries(produceInfo.info).length > 0 &&
      <Card 
        produce={produceInfo.info} 
        retailerId={selectedRetailerId}
        logo={retailersData.itemsLogos[selectedRetailerId]}
        name={producesData.itemsLabels[produceCode]}
        image={producesData.itemsImages[produceCode]}
        oneSide='true'
        noShadow={true}
      />
      }
      { youtubeId &&
        viewport !== 'mobile' &&
        <div className={s.videoContainer}>
          <h4 className={s.videoTitle}>{content.about} {producesData.itemsLabels[produceCode]}</h4>
          <p className={s.videoText}>{producesData.itemsVideosText[produceCode]}</p>
          <div className={s.video}>
            <YouTube
              videoId={youtubeId}
              id={youtubeId}
              opts={opts}
            />
          </div>
        </div>
      }
    </div>
  );
};

ProduceDetails.propTypes = {
  selectedRetailerId: PropTypes.string,
  retailersData: PropTypes.shape(),
  producesData: PropTypes.shape(),
  produceInfo: PropTypes.shape(),
  viewport: PropTypes.string,
};

ProduceDetails.defaultProps = {
  selectedRetailerId: '',
  retailersData: {},
  producesData: {},
  produceInfo: {},
  viewport: '',
};

export default ProduceDetails;
