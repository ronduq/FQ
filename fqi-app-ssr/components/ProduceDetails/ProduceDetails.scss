@import '../../style/settings/colours';
@import '../../style/settings/fonts';
@import '../../style/settings/breakpoints';

.container {
  background: white;
  border: 1px solid $colour-grey-medium;
}

.videoContainer {
  padding: 20px;
  position: relative;
  max-width: 400px;

  iframe {
    @include respond-to('two-column', min-width) {
      width: 100%;
    }
  }
}

h3.title {
  @include font-family('proxima-regular');
  @include font-size('medium');
  padding: 16px;
  border-bottom: 1px solid $colour-grey-medium;
  position: relative;
  z-index: 3;
  
  @include respond-to('tablet', min-width) {
    @include font-size('medium-20');
    padding: 5px 16px;
  }
}

h4.videoTitle {
  @include font-family('proxima-regular');
  @include font-size('medium-20');
}

.videoText {
  @include font-family('proxima-regular');
  @include font-size('small');
}
.video {
  margin-top: 20px;
}