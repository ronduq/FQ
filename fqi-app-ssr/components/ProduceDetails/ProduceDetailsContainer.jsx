import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';
import ProduceDetails from './ProduceDetails';
import { 
  getRetailersState,
  getProducesState,
  getSelectedRetailer,
  getProduceInfo,
  getViewport,
  getProduceProfileContent,
} from '../../selectors/index';
import { setViewport } from '../../actions/content';

const mapStateToProps = state => {
  return {
    retailersData: getRetailersState(state),
    producesData: getProducesState(state),
    produceInfo: getProduceInfo(state),
    selectedRetailerId: getSelectedRetailer(state),
    viewport: getViewport(state),
    content: getProduceProfileContent(state),
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setViewport: (width) => setViewport(width),
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProduceDetails);