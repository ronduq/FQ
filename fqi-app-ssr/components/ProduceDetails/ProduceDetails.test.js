/* eslint-env jest */

import React from 'react'
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer'

import ProduceDetails from './ProduceDetails'

const fakeStore = {
  dispatch: jest.fn(),
  getState: () => ({
    locations:{
      selectedLocation: {
        id: 'boston',
        label: 'Boston, MA'
      }
    },
    produceInfo: {
      info: {
        produceCode: 'test',
        id: "apple-gala",
        countriesOfOrigin: {
          '0': {countryName: "Algeria", isOrganic: true},
          '1': {countryName: "France"}
        },
        perception: "2",
        quality: "98",
        scanDate: "00/01",
        value: "2"
      }
    },
    produce: {
      id: "apple-gala",
      countriesOfOrigin: {
        '0': {countryName: "Algeria", isOrganic: true},
        '1': {countryName: "France"}
      },
      perception: "2",
      quality: "98",
      scanDate: "00/01",
      value: "2"
    },
    logo: 'https://cdn.buttercms.com/Bi087WwSQGi55NK09dsA',
    name: 'Gala Apples',
    image: "https://cdn.buttercms.com/mlIKiElrQdeZ8m7syIMx",
    cardLabels: {
      countries_of_origin: "Countries of origin",
      last_scan: "Last scan",
      organic: "Organic",
      perception: "Perception",
      product_of: "Product of",
      profile: "Profile",
      quality: "Quality",
      trends: "Trends",
      value: "Value",
    },
    producesData: {
      itemsVideos: {
        test: 'test'
      },
      itemsVideosText: {
        test: 'test'
      },
      itemsLabels: {
        test: 'test'
      },
      itemsImages: {
        test: 'test'
      },
    },
    retailersData: {
      itemsLogos: {
        test: 'test'
      }
    },
    content: {
      productCardLabels: {
        test: 'test'
      }
    }
  }),
  subscribe: jest.fn()
};

describe('With Snapshot Testing', () => {

  const props = {
    productCardLabels: {
      test: 'test'
    },
    produceInfo: {
      info: {
        produceCode: 'test',
        id: "apple-gala",
        countriesOfOrigin: {
          '0': {countryName: "Algeria", isOrganic: true},
          '1': {countryName: "France"}
        },
        perception: "2",
        quality: "98",
        scanDate: "00/01",
        value: "2"
      }
    },
    produce: {
      id: "apple-gala",
      countriesOfOrigin: {
        '0': {countryName: "Algeria", isOrganic: true},
        '1': {countryName: "France"}
      },
      perception: "2",
      quality: "98",
      scanDate: "00/01",
      value: "2"
    },
    logo: 'https://cdn.buttercms.com/Bi087WwSQGi55NK09dsA',
    name: 'Gala Apples',
    image: "https://cdn.buttercms.com/mlIKiElrQdeZ8m7syIMx",
    cardLabels: {
      countries_of_origin: "Countries of origin",
      last_scan: "Last scan",
      organic: "Organic",
      perception: "Perception",
      product_of: "Product of",
      profile: "Profile",
      quality: "Quality",
      trends: "Trends",
      value: "Value",
    },
    producesData: {
      itemsVideos: {
        test: 'test'
      },
      itemsVideosText: {
        test: 'test'
      },
      itemsLabels: {
        test: 'test'
      },
      itemsImages: {
        test: 'test'
      },
    },
    retailersData: {
      itemsLogos: {
        test: 'test'
      }
    }

  }
  it('Card snapshot"', () => {
    const component = renderer.create(
      <Provider store={fakeStore}>
        <ProduceDetails {...props} />
      </Provider>);
      const componentJSON = component.toJSON()
      expect(componentJSON).toMatchSnapshot()
  });

})