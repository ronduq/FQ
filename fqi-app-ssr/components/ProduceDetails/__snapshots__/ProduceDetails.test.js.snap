// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Card snapshot" 1`] = `
<div
  className="container"
>
  <h3
    className="title"
  >
    Produce Metrics
  </h3>
  <div
    className="card noShadow"
  >
    <div
      className="cardTop"
    >
      <img
        alt="retailer name"
        className="cardLogo"
        src=""
      />
      <img
        alt=""
        className="cardImage"
        src="test"
      />
      <h3
        className="cardTitle"
      >
        test
      </h3>
    </div>
    <div
      className="cardMidBot"
    >
      <div
        className="cardMid"
      >
        <div
          className="cardOrigin"
        >
          <svg
            aria-hidden="true"
            className="iconMedium iconTeal"
            focusable="false"
          >
            <use
              xlinkHref="#globe"
            />
          </svg>
          <span
            className="cardOriginText"
          >
            <span>
              2
            </span>
             
            n/a
          </span>
        </div>
        <div
          className="cardInfo"
        >
          <div
            className="cardInfoSide cardInfoLeft"
          >
            <div
              className="cardInfoItem"
            >
              <span
                className="cardInfoLabel"
              />
              <span
                className="cardInfoTextLarge"
              >
                98
                %
              </span>
            </div>
          </div>
          <div
            className="cardInfoSide cardInfoRight"
          >
            <div
              className="cardInfoItem"
            >
              <span
                className="cardInfoLabel"
              />
              <span
                className="cardInfoText"
              >
                2
              </span>
            </div>
            <div
              className="cardInfoItem"
            >
              <span
                className="cardInfoLabel"
              />
              <span
                className="cardInfoText"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="videoContainer"
  >
    <h4
      className="videoTitle"
    >
      About 
      test
    </h4>
    <p
      className="videoText"
    >
      test
    </p>
    <div
      className="video"
    >
      <iframe
        autoPlay="0"
        controls="0"
        frameBorder="0"
        modestbranding="1"
        rel="0"
        showinfo="0"
        src="https://www.youtube.com/embed/test"
      />
    </div>
  </div>
</div>
`;
